libats-slick/src/main/scala/com/advancedtelematic/libats/slick/db/SlickValidatedGeneric.scala

```diff
@@ -0,0 +1,17 @@
+package com.advancedtelematic.libats.slick.db
+
+import cats.syntax.either._
+import com.advancedtelematic.libats.data.ValidatedGeneric
+import slick.jdbc.MySQLProfile.api._
+
+import scala.reflect.ClassTag
+
+object SlickValidatedGeneric {
+
+  private def validatedGenericMapper[T, R: BaseColumnType](implicit gen: ValidatedGeneric[T, R], classTag: ClassTag[T]): BaseColumnType[T] =
```

Comment:
Why not make it `implicit` and kill `validatedStringMapper`?

Reply:
I tried, but then it's not enough to import `validatedGenericMapper`, I had to instantiate `implicit val m = validatedGenericMapper[EcuIdentifier, String]`.
It seems it's a limitation. From `SlickAnyVal`: 
```
// Scala implicit resolution is not smart enough to just use the method above as implicit and not use
// the `val`s below
```